Autopilot/PathManager/Inc/waypointManager.hpp

```diff
@@ -0,0 +1,236 @@
+/**
+ * Waypoint Manager Methods and Helpers
+ * Author: Dhruv Rawat
+ * Created: November 2020
+ * Last Updated: November 2020 (Dhruv)
+ */
+
+#ifndef WAYPOINT_MANAGER_H
+#define WAYPOINT_MANAGER_H
+
+#include <stdlib.h>
+#include <math.h>
+
+#define ORBIT_FOLLOWING 0
+#define LINE_FOLLOWING 1
+
+//Constants
+#define PI 3.14159265
+#define EARTH_RADIUS 6378.137
+#define PATH_BUFFER_SIZE 100
+#define MAX_PATH_APPROACH_ANGLE PI/2
+
+//Basic Mathematical Conversions
+#define deg2rad(angle_in_degrees) ((angle_in_degrees) * PI/180.0)
```

Comment:
nitpicky, and I know wer'e gonna put these in a common module anyway, but as a general rule, header files shouldn't be filled with anything the caller wouldn't care about, that's what .cpp files are for. Presenting a super clean, intuitive and simple to use interface should always be the target.